### PR TITLE
fix missing lock and de pack buf panics

### PR DESCRIPTION
**Pull request by nkunal**
# Description

1. concurrent map read/write was causing a panic during ingestion. Added missing segstore.lock
2. Don't create dependency graph if traces count is 0
3. Use cbufpool for de packing instead of a const size of 200k (fixes a panic in some scenarios)

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by nkunal** [comment]
@AndrewHess this is ready for review

File: pkg/segment/writer/segwriter.go [hunk 1]
```diff
@@ -463,12 +465,14 @@ func InitColWip(segKey string, colName string) *ColWip {
 	}
 
 	cbuf := *wipCbufPool.Get().(*[]byte)
+	dePack := *wipCbufPool.Get().(*[]byte)
```
Comment (AndrewHess) [inline pkg/segment/writer/segwriter.go:468]: You're using the same pool because `cbuf` and `dePack` should require roughly the same amount of memory, right? If they use significantly different amounts of memory, we might want to have separate pools

(this could also be changed in a different PR if we find it helps performance)

**Review by AndrewHess — APPROVED** [review]
Looks good! Just one question, but I think this can get merged as is
